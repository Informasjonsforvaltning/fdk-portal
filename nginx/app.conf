server {
    listen 8080;
    listen [::]:8080 default ipv6only=on;
    root /app;
    index index.html;
    server_name _;

    location /config.js {
        expires -1;
    }

    location / {
        try_files $uri /index.html;
    }
}
