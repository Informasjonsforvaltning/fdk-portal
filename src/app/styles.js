import 'designsystemet/fdk-designsystem-bootstrap4/scss/helper.scss';
import 'designsystemet/fdk-designsystem-bootstrap4/scss/portal.scss';
import 'designsystemet/fdk-designsystem-bootstrap4/scss/typo.scss';
import 'designsystemet/fdk-designsystem-bootstrap4/scss/common.scss';
import 'designsystemet/fdk-designsystem-bootstrap4/scss/animations.scss';
import 'designsystemet/fdk-designsystem-bootstrap4/scss/colors.scss';
import 'font-awesome/scss/font-awesome.scss';
import '../assets/fonts/icomoon.css';
