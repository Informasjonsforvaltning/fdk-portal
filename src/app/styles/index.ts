import '../../assets/designsystemet/scss/helper.scss';
import '../../assets/designsystemet/scss/portal.scss';
import '../../assets/designsystemet/scss/typo.scss';
import '../../assets/designsystemet/scss/common.scss';
import '../../assets/designsystemet/scss/animations.scss';
import '../../assets/designsystemet/scss/colors.scss';
import '../../assets/fonts/icomoon.css';

import { createGlobalStyle } from 'styled-components';

import CommonStyles from './common';

export default createGlobalStyle`
  ${CommonStyles}
`;
