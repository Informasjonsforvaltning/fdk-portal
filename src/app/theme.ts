import { Entity } from '../types/enums';

export const themeFDK = {
  colors: {
    headerBg: '#FFFFFF',
    primary: '#00C3FF',
    primaryLight: '#B3EDFF',
    primaryLighter: '#E5F9FF',
    linkDarker: '#004A73',
    linkDark: '#005484',
    link: '#0069A5',
    linkLight: '#BFD9E8',
    linkLighter: '#CCE1ED',

    neutralDarkest: '#2D3741',
    neutralDarker: '#2D3741',
    neutralDark: '#6C737A',
    neutral: '#969BA0',
    neutralLight: '#D5D7D9',
    neutralLighter: '#DFE1E2',
    neutralLightest: '#EEEFF0',
    success: '#007D69',
    successLighter: '#A6D1CA',
    successLightest: '#D9EBE8',
    warningDarker: '#885B00',
    warning: '#FFBE3C',
    warningLighter: '#FFE5B1',
    warningLightest: '#FFF2D8',
    danger: '#D70A5A',
    dangerLighter: '#F9DAE6',
    dangerLightest: '#F9DAE6',

    buttonPrimary1: '#0069A5',
    buttonPrimary2: '#005484',
    buttonSecondary1: '#CCE1ED',
    buttonSecondary2: '#BFD9E8',
    buttonShadow: '#004A73',
    buttonTextPrimary: '#FFFFFF',
    buttonTextSecondary: '#005484',
    iconStroke: '#0069A5',
    iconFill: '#CCE1ED',
    iconLink: '#0069A5',
    textDefault: '#2D3741',
    textLight: '#6C737A',
    textLink: '#0069A5',

    searchBoxBg: '#2D3741',
    searchButtonBg: '#121619',

    [Entity.DATASET]: {
      dark: '#335380',
      light: '#D5E1F2',
      lighter: '#F0F3F7'
    },
    [Entity.DATA_SERVICE]: {
      dark: '#805333',
      light: '#F2E1D5',
      lighter: '#F7F2ED'
    },
    [Entity.CONCEPT]: {
      dark: '#2E6773',
      light: '#D5EDF2',
      lighter: '#F0F6F7'
    },
    [Entity.INFORMATION_MODEL]: {
      dark: '#593380',
      light: '#E4D5F2',
      lighter: '#F4F0F7'
    }
  }
};

export const themeNAP = {
  colors: {
    header: '#FFFFFF',
    headerBg: '#444F55',
    primary: '#ED9300',
    primaryLight: '#FADFB2',
    primaryLighter: '#FDF4E5',
    linkDarker: '#8E5800',
    linkDark: '#D58400',
    link: '#ED9300',
    linkLight: '#F9D499',
    linkLighter: '#FAE4BF',

    neutralDarkest: '#2E3539',
    neutralDarker: '#2E3439',
    neutralDark: '#444F55',
    neutral: '#A1A7AA',
    neutralLight: '#DADADA',
    neutralLighter: '#E5E5E5',
    neutralLightest: '#F2F2F2',
    success: '#3E7B1F',
    successLighter: '#ABD795',
    successLightest: '#EBF6E6',
    warningDarker: '#8E5800',
    warning: '#ED9300',
    warningLighter: '#FFE5B1',
    warningLightest: '#FFF2D8',
    danger: '#C83214',
    dangerLighter: '#ECB7AD',
    dangerLightest: '#F7E0DC',

    buttonPrimary1: '#008ec2',
    buttonPrimary2: '#F1A933',
    buttonSecondary1: '#F8D499',
    buttonSecondary2: '#FAE4BF',
    buttonShadow: '#8E5800',
    buttonTextPrimary: '#FFFFFF',
    buttonTextSecondary: '#2E3539',
    iconStroke: '#2E3539',
    iconFill: '#ED9300',
    iconLink: '#444F55',
    textDefault: '#444F55',
    textLight: '#60696E',
    textLink: '#444F55',

    searchBoxBg: '#DFE1E2',
    searchButtonBg: '#D5D7D9',

    [Entity.DATASET]: {
      dark: '#3E7B1F',
      light: '#DFF2D5',
      lighter: '#F4F7F2'
    },
    [Entity.DATA_SERVICE]: {
      dark: '#805333',
      light: '#F2E1D5',
      lighter: '#F7F2ED'
    },
    [Entity.CONCEPT]: {
      dark: '#337380',
      light: '#D5EDF2',
      lighter: '#F0F6F7'
    },
    [Entity.INFORMATION_MODEL]: {
      dark: '#593380',
      light: '#E4D5F2',
      lighter: '#F4F0F7'
    }
  }
};
