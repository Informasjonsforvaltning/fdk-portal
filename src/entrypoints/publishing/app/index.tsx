import React, { FC } from 'react';
import { hot } from 'react-hot-loader/root';
import { Provider as ReduxProvider } from 'react-redux';
import ThemeProvider from '@fellesdatakatalog/theme';

import TranslationsProvider from '../../../providers/translations';

import store from '../redux/store';

import GlobalStyles from '../styles';
import Router from '../router';

import AnalyticsGA from '../../../components/analytics-ga';
import AnalyticsMonsido from '../../../components/analytics-monsido';
import AnalyticsSiteImprove from '../../../components/analytics-siteimprove';

const App: FC = () => (
  <ThemeProvider>
    <GlobalStyles />
    <AnalyticsGA />
    <AnalyticsMonsido />
    <AnalyticsSiteImprove />
    <TranslationsProvider>
      <ReduxProvider store={store}>
        <Router />
      </ReduxProvider>
    </TranslationsProvider>
  </ThemeProvider>
);

export default hot(App);
