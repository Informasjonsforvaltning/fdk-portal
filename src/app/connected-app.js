import { connect } from 'react-redux';
import { App } from './app';

const mapStateToProps = state => ({
  // react-localization module designed to be stateful, so we have to re-render the app when the selected language is changed
  language: state.settings.language
});

export const ConnectedApp = connect(mapStateToProps)(App);
