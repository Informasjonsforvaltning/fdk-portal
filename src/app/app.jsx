import React from 'react';
import { Route, Switch } from 'react-router-dom';
import PropTypes from 'prop-types';
import cx from 'classnames';
import DocumentMeta from 'react-document-meta';

import localization from '../lib/localization';
import { ConnectedSearchPage } from '../pages/search-page/connected-search-page';
import { DatasetDetailsPage } from '../pages/dataset-details-page/dataset-details-page';
import { ConnectedApiDetailsPage } from '../pages/api-details-page/connected-api-details-page';
import { ConnectedConceptDetailsPage } from '../pages/concept-details-page/connected-concept-details-page';
import { ConnectedConceptComparePage } from '../pages/concept-compare-page/connected-concept-compare-page';
import { ConnectedInformationModelDetailsPage } from '../pages/informationmodel-details-page/connected-information-model-details-page';
import { AboutPage } from '../pages/about-page/about-page.component';
import { ArticlePage } from '../pages/article-page/article-page.component';
import { Breadcrumbs } from './breadcrumbs/breadcrumbs.component';
import { ConnectedAppNavBar } from './app-nav-bar/connected-app-nav-bar';
import { ConnectedDatasetsReportPage } from '../pages/datasets-report-page/connected-datasets-report-page';
import {
  PATHNAME_DATASETS,
  PATHNAME_DATASET_DETAILS,
  PATHNAME_APIS,
  PATHNAME_CONCEPTS,
  PATHNAME_CONCEPTS_COMPARE,
  PATHNAME_INFORMATIONMODELS,
  PATHNAME_REPORTS,
  PATHNAME_ABOUT,
  PATHNAME_ABOUT_REGISTRATION
} from '../constants/constants';
import ScrollToTop from '../components/scroll-to-top/scrollToTop.component';
import { getConfig } from '../config';
import '../assets/css/bootstrap-override.scss';
import './styles';

export function App({ language }) {
  // react-localization is a stateful library, so we set the required language on each full-app render
  // and full-render app each time when the language is changed
  localization.setLanguage(language);
  const themeClass = cx({
    'theme-nap': getConfig().themeNap,
    'theme-fdk': !getConfig().themeNap
  });

  const footerText = !getConfig().themeNap
    ? localization.footer.information_text
    : localization.footer.information_textNap;

  const footerEmail = !getConfig().themeNap
    ? localization.footer.mail
    : localization.footer.mailNap;

  return (
    <div className={themeClass}>
      {getConfig().themeNap && <DocumentMeta {...{ title: 'NAP' }} />}
      <div>
        <a
          id="focus-element"
          className="uu-invisible"
          href={`${location.pathname}#content`}
          aria-hidden="true"
        >
          Hopp til hovedinnhold
        </a>
      </div>
      <div id="skip-link-wrap">
        <a id="skip-link" href={`${location.pathname}#content`}>
          Hopp til hovedinnhold
        </a>
      </div>

      <ConnectedAppNavBar />

      <div className="container">
        <div className="row">
          <div className="col-12">
            <Breadcrumbs />
          </div>
        </div>
      </div>
      <div className="app-routes">
        <Switch>
          <Route
            exact
            path={PATHNAME_DATASETS}
            component={ConnectedSearchPage}
          />
          <Route exact path={PATHNAME_APIS} component={ConnectedSearchPage} />
          <Route
            exact
            path={PATHNAME_CONCEPTS}
            component={ConnectedSearchPage}
          />
          <Route
            exact
            path={PATHNAME_INFORMATIONMODELS}
            component={ConnectedSearchPage}
          />
          <ScrollToTop>
            <Switch>
              <Route
                exact
                path={`${PATHNAME_INFORMATIONMODELS}/:id`}
                component={ConnectedInformationModelDetailsPage}
              />
              <Route
                exact
                path={`${PATHNAME_CONCEPTS}${PATHNAME_CONCEPTS_COMPARE}`}
                component={ConnectedConceptComparePage}
              />
              <Route
                exact
                path={`${PATHNAME_CONCEPTS}/:id`}
                component={ConnectedConceptDetailsPage}
              />
              <Route
                exact
                path={`${PATHNAME_DATASET_DETAILS}/:id`}
                component={DatasetDetailsPage}
              />
              <Route
                exact
                path="/apis/:id"
                component={ConnectedApiDetailsPage}
              />
              <Route
                exact
                path={PATHNAME_REPORTS}
                component={ConnectedDatasetsReportPage}
              />
              <Route exact path={PATHNAME_ABOUT} component={AboutPage} />
              <Route
                exact
                path={PATHNAME_ABOUT_REGISTRATION}
                component={ArticlePage}
              />
            </Switch>
          </ScrollToTop>
        </Switch>
      </div>

      <div className="fdk-footer d-md-none">
        <div className="container">
          <div className="row">
            <div className="col-sm-12 text-center mb-2">
              <p className="fdk-p-footer">{footerText}</p>
            </div>
            <div className="col-sm-12 text-center mb-2">
              <p className="fdk-p-footer">
                <a href="https://www.brreg.no/personvernerklaering/">
                  {localization.footer.information}
                  {localization.footer.privacy}
                  <i className="fa fa-external-link fdk-fa-right" />
                </a>
              </p>
            </div>

            <div className="col-sm-12 text-center mb-2">
              <p className="fdk-p-footer">
                <a href={`mailto:${footerEmail}`}>{footerEmail}</a>
              </p>
            </div>
          </div>
        </div>
      </div>

      <div className="fdk-footer d-none d-md-block">
        <div className="container">
          <div className="row">
            <div className="col-md-3">
              <p className="fdk-p-footer">
                <a href="https://www.brreg.no/personvernerklaering/">
                  {localization.footer.information}
                  <br />
                  {localization.footer.privacy}
                  <i className="fa fa-external-link fdk-fa-right" />
                </a>
              </p>
            </div>
            <div className="col-md-6 text-center">
              <span className="uu-invisible" aria-hidden="false">
                Felles Datakatalog.
              </span>
              <p className="fdk-p-footer">{footerText}</p>
            </div>
            <div className="col-md-3 text-right">
              <p className="fdk-p-footer">
                <a href={`mailto:${footerEmail}`}>
                  <span className="uu-invisible" aria-hidden="false">
                    Mailadresse.
                  </span>
                  {localization.footer.contact}
                  <br />
                  {footerEmail}
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

App.propTypes = {
  language: PropTypes.string.isRequired
};
