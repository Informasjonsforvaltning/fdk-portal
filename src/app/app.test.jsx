import React from 'react';
import { shallow } from 'enzyme';
import { App } from './app';

test('should render App correctly', () => {
  const wrapper = shallow(<App language="nb" />);
  expect(wrapper).toMatchSnapshot();
});
