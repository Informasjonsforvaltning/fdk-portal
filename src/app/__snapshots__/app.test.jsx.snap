// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App correctly 1`] = `
<div
  className="theme-fdk"
>
  <div>
    <a
      aria-hidden="true"
      className="uu-invisible"
      href="/#content"
      id="focus-element"
    >
      Hopp til hovedinnhold
    </a>
  </div>
  <div
    id="skip-link-wrap"
  >
    <a
      href="/#content"
      id="skip-link"
    >
      Hopp til hovedinnhold
    </a>
  </div>
  <Connect(AppNavBar) />
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <withRouter() />
      </div>
    </div>
  </div>
  <div
    className="app-routes"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/apis"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/concepts"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/informationmodels"
      />
      <withRouter(ScrollToTop)>
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/informationmodels/:id"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/concepts/compare"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/concepts/:id"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/datasets/:id"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/apis/:id"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/reports"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/about"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/about-registration"
          />
        </Switch>
      </withRouter(ScrollToTop)>
    </Switch>
  </div>
  <div
    className="fdk-footer d-md-none"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-sm-12 text-center mb-2"
        >
          <p
            className="fdk-p-footer"
          >
            Dette er et samarbeidsprosjekt mellom Brønnøysundregistrene, Difi og øvrige SKATE-etater.
          </p>
        </div>
        <div
          className="col-sm-12 text-center mb-2"
        >
          <p
            className="fdk-p-footer"
          >
            <a
              href="https://www.brreg.no/personvernerklaering/"
            >
              Informasjonskapsler
               og personvern
              <i
                className="fa fa-external-link fdk-fa-right"
              />
            </a>
          </p>
        </div>
        <div
          className="col-sm-12 text-center mb-2"
        >
          <p
            className="fdk-p-footer"
          >
            <a
              href="mailto:fellesdatakatalog@brreg.no"
            >
              fellesdatakatalog@brreg.no
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="fdk-footer d-none d-md-block"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-3"
        >
          <p
            className="fdk-p-footer"
          >
            <a
              href="https://www.brreg.no/personvernerklaering/"
            >
              Informasjonskapsler
              <br />
               og personvern
              <i
                className="fa fa-external-link fdk-fa-right"
              />
            </a>
          </p>
        </div>
        <div
          className="col-md-6 text-center"
        >
          <span
            aria-hidden="false"
            className="uu-invisible"
          >
            Felles Datakatalog.
          </span>
          <p
            className="fdk-p-footer"
          >
            Dette er et samarbeidsprosjekt mellom Brønnøysundregistrene, Difi og øvrige SKATE-etater.
          </p>
        </div>
        <div
          className="col-md-3 text-right"
        >
          <p
            className="fdk-p-footer"
          >
            <a
              href="mailto:fellesdatakatalog@brreg.no"
            >
              <span
                aria-hidden="false"
                className="uu-invisible"
              >
                Mailadresse.
              </span>
              Kontakt:
              <br />
              fellesdatakatalog@brreg.no
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
